src/oscar/apps/basket/views.py

```diff
@@ -326,12 +326,7 @@ def get_form_kwargs(self):
         return kwargs
 
     def form_invalid(self, form):
-        msgs = []
-        for error in form.errors.values():
-            msgs.append(error.as_text())
-        clean_msgs = [m.replace('* ', '') for m in msgs if m.startswith('* ')]
-        messages.error(self.request, ",".join(clean_msgs))
-
+        self.request.session["add_to_basket_form_post_data"] = self.request.POST
```

Comment:
Since you are now storing arbitrary posted data in the session, you are open for a remote execution attack:

https://docs.djangoproject.com/en/3.2/topics/http/sessions/#cookie-session-backend

Please add a check that settings.SESSION_SERIALIZER has been set to ``django.contrib.sessions.serializers.JSONSerializer`` and refuse to store the POST data in the session if not so. Also add a warning.

Reply:
maybe use the existing functionality if an unsafe session serializer has been set

Reply:
okay added the check